### DISCUSSION TITLE
How to understand gsoc-first issues?

### DISCUSSION THREAD

**Discussion by nitishfy** [checkstyle/checkstyle#12415] [Q&A] [answered]
Right now, I'm going through the closed gsoc-first issues which have been solved by folks.

I'm not able to understand how do people know what code to write and what not to in order to add it in examples?

Could anyone share their approach on how do you go about solving such issues where code-example needs to be added?

**Comment by nrmancuso** [comment, marked as answer]
1. Read existing documentation 
2. Look through unit tests for the check
3. Create and play with examples locally to understand check

**Reply by nitishfy** [reply]
Does this mean that I need to create a new input file (to understand a check) in separate windows and write code which violates that check?
Once the violation has been raised, I need to make some changes to the code and push it in the form of PR, right?

**Reply by nrmancuso** [reply]
Follow the bug report template pattern to play with and understand checks using all different properties from existing documentation

**Reply by nitishfy** [reply]
Is this what you're mentioning about? https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

**Reply by nrmancuso** [reply]
https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE/bug_report.md
